app/controllers/nodes_controller.rb

```diff
@@ -18,18 +18,34 @@ def edit
   # POST /nodes
   def create
     @node.label = 'unnamed' unless @node.label.present?
-    if @node.save
-      track_created(@node)
-      flash[:notice] = 'Successfully created node.'
-      redirect_to @node
-    else
-      parent = @node.parent
-      if parent
-        redirect_to parent, alert: @node.errors.full_messages.join('; ')
-      else
-        redirect_to summary_path, alert: @node.errors.full_messages.join('; ')
+    @node.save!
```

Comment:
Yes, but the conditional logic is redundant because the node will never be invalid (i.e. the `else` block never gets called.) There's only one validation on `Node`, a presence validation on `label`. But that will never fail in this case because nodes are given a default label on [line 20](https://github.com/dradis/dradis-ce/pull/40/files#diff-f6042ba19178936a392b7082cf00c9e5L20).

In fact, I'll add a comment to this method to make this clearer.